lib/internal/test_runner/test.js

```diff
@@ -137,7 +138,8 @@ class Test extends AsyncResource {
       this.concurrency = concurrency;
     }
 
-    if (isUint32(timeout)) {
+    if (timeout != null && timeout !== Infinity) {
+      validateUint32(timeout, 'options.timeout');
```

Comment:
Is there a reason for requiring an `uint32` here instead of any positive (in particular, non-`NaN`) `number`?

For example, because timeouts are inherently flaky, it is often necessary to specify a factor depending on the platform to reduce flakiness (as we do for many platforms). It might be surprising if `1.5 * timeout` suddenly fails because it's not an `uint32`.

Reply:
the reason is that is what timeouts support https://github.com/nodejs/node/blob/92567283f4027940d4a4df0e7fb7444b67842ed5/lib/internal/timers.js#L167-L172

Reply:
@MoLow But timeouts allow non-integers in the range `[0, TIMEOUT_MAX]`. For example, let's say a `timeout` is `9999` and someone pre-scales it with a factor of `1.5` on some slow platform because the test is flaky. The result is `14998.5`, which is a valid timeout, but not an uint32.